addons/account/wizard/account_automatic_entry_wizard.py

```diff
@@ -207,10 +207,16 @@ def _get_move_dict_vals_change_account(self):
                     'amount_currency': (account_balance > 0 and -1 or 1) * abs(account_amount_currency),
                 })
 
+        # Get the lowest child company based on accounts used to avoid access error
+        accounts = self.env['account.account'].browse([line['account_id'] for line in line_vals])
+        companies = accounts.company_id | self.env.company
+        lowest_child_company = companies.sorted(lambda company: len(company.parent_ids))[-1]
```

Comment:
Just some nitpick: `sorted` is `n*log(n)` (+ uses memory), so it is generally better to use `max()` or `min()` with a lambda to stay in `n` instead of sorting and taking the first/last.
Is fine on companies since there will never be more than a few.

Reply:
Yes you are right, and it will be more readable.